modules/compiler/utils/source/replace_wgc_pass.cpp

```diff
@@ -615,6 +615,11 @@ PreservedAnalyses compiler::utils::ReplaceWGCPass::run(
     auto Builtin = BI.analyzeBuiltin(F);
     if (auto WGC = BI.isMuxGroupCollective(Builtin.ID);
         WGC && WGC->isWorkGroupScope()) {
+      if (scans_only &&
```

Comment:
`scans_only && !WGC->isScan()`  would be simpler

Reply:
good call